### ISSUE TITLE
[BUG] `yolov8` pretrained weights are not matching with yaml

### ISSUE THREAD

**Issue by parthadipxyz** [ultralytics/ultralytics#16678]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

So, I am training a yolov8n model in coco2017 dataset throughout the last month. I have trained it upto 31 epoch and after that when I tried to train it for the 32th epoch, I saw that at the beginning of training a the message "Transfered 381/453 items from pretrained weight" is shown and amp is checked with YOLO11n model. The model at 31 epoch and before that detects objects in images just fine but the model after 32th epoch shows no detection for the same or different image. I haven't changed anything about the model or dataset configuration as far as I can remember but I am suspicious that maybe yolo v11 realease somehow broke something somewhere. Another thing to note is that, during 32th epoch the losses are comparable to that of the 31th epoch yet the 32th one and even the one's 3 or 4 epoch after that detects nothing. Again from 32 to 34 and onwards I can see "Transfered 453/453 items from pretrained weight". So that makes me suspect the changes somewhere else on the repo as the reason. I'm not sure about it. I would really appreciate it if someone could point out something that I can try.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @parthadipxyz, thank you for reaching out to Ultralytics 🚀! We understand you're experiencing an issue with your YOLOv8 training after 31 epochs.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to facilitate debugging.

For custom training ❓ questions, please provide detailed information, such as dataset examples, training logs, and ensure you're adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

We suggest checking out our [Docs](https://docs.ultralytics.com) for common solutions or exploring [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

Engage with the Ultralytics community! Join our [Discord](https://ultralytics.com/discord) 🎧 for real-time discussions, visit [Discourse](https://community.ultralytics.com) for in-depth topics, or engage on [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Please ensure your environment is up to date with the latest package and requirements:

```bash
pip install -U ultralytics
```

## Environments

Run YOLO in verified environments with all dependencies preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates all CI tests are passing. These tests verify operations across platforms for YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

This is an automated response; an Ultralytics engineer will assist you soon. 🎉

**Comment by Y-T-G** [comment]
Provide the training command or code. Are you resuming or restarting training?

**Comment by parthadipxyz** [comment]
model = YOLO("/content/drive/MyDrive/Research/yolov8n-seg.yaml").load(model_path)  # build from YAML and transfer weights

project_dir="/content/drive/MyDrive/Research/runs"+next_iteration
model.train(data="coco8-seg.yaml",optimizer='SGD',imgsz=640,cls=0.0,batch=42,epochs=1,project=project_dir)


I am training in each epoch using the last.pt from previous epoch

**Comment by Y-T-G** [comment]
When you restart training like that, the learning rate is reset and it applies warmup which is a very high learning rate.

https://redd.it/1eolwl8

**Comment by Y-T-G** [comment]
> model = YOLO("/content/drive/MyDrive/Research/yolov8n-seg.yaml").load(model_path)  # build from YAML and transfer weights
> 

When you run this, how many weights are transferred?

**Comment by parthadipxyz** [comment]
Even with that at the beginning 381 items are transfrred but the model in yaml and the latest model has no difference in architecture

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by Y-T-G** [comment]
@ambitious-octopus I can reproduce this.

```python
from ultralytics import YOLO

model = YOLO("yolov8n-seg.yaml").load("yolov8n-seg.pt")
``` 

`Transferred 381/453 items from pretrained weights`

**Comment by parthadipxyz** [comment]
Well then where is the actually issue?

**Comment by Y-T-G** [comment]
Occurring with `Segment`, `OBB` and `Pose`.


```
Ultralytics 8.3.4 🚀 Python-3.10.14 torch-2.4.1+cu121 CUDA:0 (Tesla T4, 14916MiB)
Setup complete ✅ (64 CPUs, 433.0 GB RAM, 739.4/992.2 GB disk)

OS                  Linux-5.15.0-1017-azure-x86_64-with-glibc2.35
Environment         Docker
Python              3.10.14
Install             git
RAM                 433.01 GB
CPU                 AMD EPYC 7V12 64-Core Processor
CUDA                12.1

numpy               ✅ 1.23.5<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.6.0.66>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.28.2>=2.23.0
scipy               ✅ 1.14.0>=1.4.1
torch               ✅ 2.4.1>=1.8.0
torchvision         ✅ 0.19.1>=0.9.0
tqdm                ✅ 4.65.2>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0
torch               ✅ 2.4.1!=2.4.0,>=1.8.0; sys_platform == "win32"
```

**Comment by Y-T-G** [comment]
The last layer keys are not matching:

```python
model = YOLO('yolov8n-seg.pt')
ptk = model.model.model.state_dict().keys()

model = YOLO('yolov8n-seg.yaml')
yamlk = model.model.model.state_dict().keys()

for k1, k2 in zip(yamlk, ptk):
    if k1 != k2:
        print(k1, "!=", k2)

22.cv3.0.0.0.conv.weight != 22.cv3.0.0.conv.weight
22.cv3.0.0.0.bn.weight != 22.cv3.0.0.bn.weight
22.cv3.0.0.0.bn.bias != 22.cv3.0.0.bn.bias
22.cv3.0.0.0.bn.running_mean != 22.cv3.0.0.bn.running_mean
22.cv3.0.0.0.bn.running_var != 22.cv3.0.0.bn.running_var
22.cv3.0.0.0.bn.num_batches_tracked != 22.cv3.0.0.bn.num_batches_tracked
22.cv3.0.0.1.conv.weight != 22.cv3.0.1.conv.weight
22.cv3.0.0.1.bn.weight != 22.cv3.0.1.bn.weight
22.cv3.0.0.1.bn.bias != 22.cv3.0.1.bn.bias
22.cv3.0.0.1.bn.running_mean != 22.cv3.0.1.bn.running_mean
22.cv3.0.0.1.bn.running_var != 22.cv3.0.1.bn.running_var
22.cv3.0.0.1.bn.num_batches_tracked != 22.cv3.0.1.bn.num_batches_tracked
22.cv3.0.1.0.conv.weight != 22.cv3.0.2.weight
22.cv3.0.1.0.bn.weight != 22.cv3.0.2.bias
22.cv3.0.1.0.bn.bias != 22.cv3.1.0.conv.weight
22.cv3.0.1.0.bn.running_mean != 22.cv3.1.0.bn.weight
22.cv3.0.1.0.bn.running_var != 22.cv3.1.0.bn.bias
22.cv3.0.1.0.bn.num_batches_tracked != 22.cv3.1.0.bn.running_mean
22.cv3.0.1.1.conv.weight != 22.cv3.1.0.bn.running_var
22.cv3.0.1.1.bn.weight != 22.cv3.1.0.bn.num_batches_tracked
22.cv3.0.1.1.bn.bias != 22.cv3.1.1.conv.weight
22.cv3.0.1.1.bn.running_mean != 22.cv3.1.1.bn.weight
22.cv3.0.1.1.bn.running_var != 22.cv3.1.1.bn.bias
22.cv3.0.1.1.bn.num_batches_tracked != 22.cv3.1.1.bn.running_mean
22.cv3.0.2.weight != 22.cv3.1.1.bn.running_var
22.cv3.0.2.bias != 22.cv3.1.1.bn.num_batches_tracked
22.cv3.1.0.0.conv.weight != 22.cv3.1.2.weight
22.cv3.1.0.0.bn.weight != 22.cv3.1.2.bias
22.cv3.1.0.0.bn.bias != 22.cv3.2.0.conv.weight
22.cv3.1.0.0.bn.running_mean != 22.cv3.2.0.bn.weight
22.cv3.1.0.0.bn.running_var != 22.cv3.2.0.bn.bias
22.cv3.1.0.0.bn.num_batches_tracked != 22.cv3.2.0.bn.running_mean
22.cv3.1.0.1.conv.weight != 22.cv3.2.0.bn.running_var
22.cv3.1.0.1.bn.weight != 22.cv3.2.0.bn.num_batches_tracked
22.cv3.1.0.1.bn.bias != 22.cv3.2.1.conv.weight
22.cv3.1.0.1.bn.running_mean != 22.cv3.2.1.bn.weight
22.cv3.1.0.1.bn.running_var != 22.cv3.2.1.bn.bias
22.cv3.1.0.1.bn.num_batches_tracked != 22.cv3.2.1.bn.running_mean
22.cv3.1.1.0.conv.weight != 22.cv3.2.1.bn.running_var
22.cv3.1.1.0.bn.weight != 22.cv3.2.1.bn.num_batches_tracked
22.cv3.1.1.0.bn.bias != 22.cv3.2.2.weight
22.cv3.1.1.0.bn.running_mean != 22.cv3.2.2.bias
22.cv3.1.1.0.bn.running_var != 22.dfl.conv.weight
22.cv3.1.1.0.bn.num_batches_tracked != 22.proto.cv1.conv.weight
22.cv3.1.1.1.conv.weight != 22.proto.cv1.bn.weight
22.cv3.1.1.1.bn.weight != 22.proto.cv1.bn.bias
22.cv3.1.1.1.bn.bias != 22.proto.cv1.bn.running_mean
22.cv3.1.1.1.bn.running_mean != 22.proto.cv1.bn.running_var
22.cv3.1.1.1.bn.running_var != 22.proto.cv1.bn.num_batches_tracked
22.cv3.1.1.1.bn.num_batches_tracked != 22.proto.upsample.weight
22.cv3.1.2.weight != 22.proto.upsample.bias
22.cv3.1.2.bias != 22.proto.cv2.conv.weight
22.cv3.2.0.0.conv.weight != 22.proto.cv2.bn.weight
22.cv3.2.0.0.bn.weight != 22.proto.cv2.bn.bias
22.cv3.2.0.0.bn.bias != 22.proto.cv2.bn.running_mean
22.cv3.2.0.0.bn.running_mean != 22.proto.cv2.bn.running_var
22.cv3.2.0.0.bn.running_var != 22.proto.cv2.bn.num_batches_tracked
22.cv3.2.0.0.bn.num_batches_tracked != 22.proto.cv3.conv.weight
22.cv3.2.0.1.conv.weight != 22.proto.cv3.bn.weight
22.cv3.2.0.1.bn.weight != 22.proto.cv3.bn.bias
22.cv3.2.0.1.bn.bias != 22.proto.cv3.bn.running_mean
22.cv3.2.0.1.bn.running_mean != 22.proto.cv3.bn.running_var
22.cv3.2.0.1.bn.running_var != 22.proto.cv3.bn.num_batches_tracked
22.cv3.2.0.1.bn.num_batches_tracked != 22.cv4.0.0.conv.weight
22.cv3.2.1.0.conv.weight != 22.cv4.0.0.bn.weight
22.cv3.2.1.0.bn.weight != 22.cv4.0.0.bn.bias
22.cv3.2.1.0.bn.bias != 22.cv4.0.0.bn.running_mean
22.cv3.2.1.0.bn.running_mean != 22.cv4.0.0.bn.running_var
22.cv3.2.1.0.bn.running_var != 22.cv4.0.0.bn.num_batches_tracked
22.cv3.2.1.0.bn.num_batches_tracked != 22.cv4.0.1.conv.weight
22.cv3.2.1.1.conv.weight != 22.cv4.0.1.bn.weight
22.cv3.2.1.1.bn.weight != 22.cv4.0.1.bn.bias
22.cv3.2.1.1.bn.bias != 22.cv4.0.1.bn.running_mean
22.cv3.2.1.1.bn.running_mean != 22.cv4.0.1.bn.running_var
22.cv3.2.1.1.bn.running_var != 22.cv4.0.1.bn.num_batches_tracked
22.cv3.2.1.1.bn.num_batches_tracked != 22.cv4.0.2.weight
22.cv3.2.2.weight != 22.cv4.0.2.bias
22.cv3.2.2.bias != 22.cv4.1.0.conv.weight
22.dfl.conv.weight != 22.cv4.1.0.bn.weight
22.proto.cv1.conv.weight != 22.cv4.1.0.bn.bias
22.proto.cv1.bn.weight != 22.cv4.1.0.bn.running_mean
22.proto.cv1.bn.bias != 22.cv4.1.0.bn.running_var
22.proto.cv1.bn.running_mean != 22.cv4.1.0.bn.num_batches_tracked
22.proto.cv1.bn.running_var != 22.cv4.1.1.conv.weight
22.proto.cv1.bn.num_batches_tracked != 22.cv4.1.1.bn.weight
22.proto.upsample.weight != 22.cv4.1.1.bn.bias
22.proto.upsample.bias != 22.cv4.1.1.bn.running_mean
22.proto.cv2.conv.weight != 22.cv4.1.1.bn.running_var
22.proto.cv2.bn.weight != 22.cv4.1.1.bn.num_batches_tracked
22.proto.cv2.bn.bias != 22.cv4.1.2.weight
22.proto.cv2.bn.running_mean != 22.cv4.1.2.bias
22.proto.cv2.bn.running_var != 22.cv4.2.0.conv.weight
22.proto.cv2.bn.num_batches_tracked != 22.cv4.2.0.bn.weight
22.proto.cv3.conv.weight != 22.cv4.2.0.bn.bias
22.proto.cv3.bn.weight != 22.cv4.2.0.bn.running_mean
22.proto.cv3.bn.bias != 22.cv4.2.0.bn.running_var
22.proto.cv3.bn.running_mean != 22.cv4.2.0.bn.num_batches_tracked
22.proto.cv3.bn.running_var != 22.cv4.2.1.conv.weight
22.proto.cv3.bn.num_batches_tracked != 22.cv4.2.1.bn.weight
22.cv4.0.0.conv.weight != 22.cv4.2.1.bn.bias
22.cv4.0.0.bn.weight != 22.cv4.2.1.bn.running_mean
22.cv4.0.0.bn.bias != 22.cv4.2.1.bn.running_var
22.cv4.0.0.bn.running_mean != 22.cv4.2.1.bn.num_batches_tracked
22.cv4.0.0.bn.running_var != 22.cv4.2.2.weight
```

**Comment by parthadipxyz** [comment]
How to solve that?

**Comment by Y-T-G** [comment]
> How to solve that?

You can install older ultralytics:
`pip install "ultralytics<8.3.0`

**Comment by parthadipxyz** [comment]
Issue solved. Thanks a lot

**Comment by Y-T-G** [comment]
Reopening since it's still a bug in the latest version.

**Comment by ambitious-octopus** [comment]
Created fix PR: https://github.com/ultralytics/ultralytics/pull/16679

**Comment by Laughing-q** [comment]
@parthadipxyz @Y-T-G @ambitious-octopus Oh yes because we updated the Detect head for yolo11 models so every training launched from `yolov8.yaml` would be different from previous version.
Also I tested current version is also compatible with the v8 models trained before the yolo11 release version 8.3.0, which means we can still train/predict/val/export with pretrained yolov8 weights or any other custom weights trained before 8.3.0. 
But yes if we sre going to start a training from yaml file then the Detect head would be created as now yolo11's head.
@parthadipxyz As for your case you want to resume a training, which internally we are creating a model by model.yaml then load the weight, so it creates from the new yolo11 head. And the solution for now I suggest is to rollback the package version to `<8.3.0`.

**Comment by Y-T-G** [comment]
We probably need a v11 counterpart for Pose, Segment and OBB heads to inherit from `v11Detect`.

```
In [1]: from ultralytics import YOLO
model
In [2]: model = YOLO('yolo11n-seg.yaml').load('yolo11n-seg.pt')
Transferred 489/525 items from pretrained weights
```

**Comment by glenn-jocher** [comment]
@Laughing-q so this looks like users will get a hybrid YOLOv8-YOLO11 model then if they use yolov8.yaml with 8.3.0 right? 

Maybe we should think if how to split the two better so there's less confusion.

**Comment by Laughing-q** [comment]
> so this looks like users will get a hybrid YOLOv8-YOLO11 model then if they use yolov8.yaml with 8.3.0 right?

@glenn-jocher yes

**Comment by parthadipxyz** [comment]
Wouldn't it be better (if possible) to integrate everything related to the architecture of a specific model within the yaml file of that model? Is there any specific benefit of keeping the head as a separate entity? I am just curious.

**Comment by glenn-jocher** [comment]
The separation of the model head from the YAML file allows for greater flexibility and modularity, enabling easier updates and customization of different components without altering the entire architecture.
